@xen-orchestra/lite/src/components/RemoteConsole.vue

```diff
@@ -3,42 +3,96 @@
 </template>
 
 <script lang="ts" setup>
-import { onBeforeUnmount, ref, watchEffect } from "vue";
+import { fibonacci } from "iterable-backoff";
+import { computed, onBeforeUnmount, ref, watch, watchEffect } from "vue";
 import VncClient from "@novnc/novnc/core/rfb";
 import { useXenApiStore } from "@/stores/xen-api.store";
 
+const N_TOTAL_TRIES = 8;
+const FIBONACCI_MS_ARRAY: number[] = Array.from(
+  fibonacci().toMs().take(N_TOTAL_TRIES)
+);
+const INITIAL_VALUE_N_CONNECTION_ATTEMPTS = 0;
+
 const props = defineProps<{
   location: string;
+  isConsoleAvailable: boolean;
 }>();
 
 const vmConsoleContainer = ref<HTMLDivElement>();
 const xenApiStore = useXenApiStore();
-let vncClient: VncClient | undefined;
-
-watchEffect(() => {
-  if (!vmConsoleContainer.value || !xenApiStore.currentSessionId) {
+const url = computed(() => {
+  if (xenApiStore.currentSessionId === null) {
     return;
   }
+  const _url = new URL(props.location);
+  _url.protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
+  _url.searchParams.set("session_id", xenApiStore.currentSessionId);
+  return _url;
+});
 
-  if (vncClient !== undefined) {
-    vncClient.disconnect();
-    vncClient = undefined;
+let vncClient: VncClient | undefined;
+let nConnectionAttempts = INITIAL_VALUE_N_CONNECTION_ATTEMPTS;
+
+const handleDisconnectionEvent = () => {
+  clearVncClient();
+  if (props.isConsoleAvailable) {
+    console.error(
+      `Connection lost for the remote console: ${
+        props.location
+      }. New attempt in ${FIBONACCI_MS_ARRAY[nConnectionAttempts - 1]}ms`
+    );
+    createVncConnection();
   }
+};
+const handleConnectionEvent = () =>
+  (nConnectionAttempts = INITIAL_VALUE_N_CONNECTION_ATTEMPTS);
 
-  const url = new URL(props.location);
-  url.protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
-  url.searchParams.set("session_id", xenApiStore.currentSessionId);
+const clearVncClient = () => {
+  vncClient?.removeEventListener("disconnect", handleDisconnectionEvent);
+  vncClient?.removeEventListener("connect", handleConnectionEvent);
+  if (vncClient?._rfbConnectionState !== "disconnected") {
+    vncClient?.disconnect();
+  }
+  vncClient = undefined;
+};
+const createVncConnection = async () => {
+  if (nConnectionAttempts > N_TOTAL_TRIES) {
+    console.error(
+      "The number of reconnection attempts has been exceeded for:",
+      props.location
+    );
+    return;
+  }
+  await new Promise((resolve) =>
+    setTimeout(resolve, FIBONACCI_MS_ARRAY[nConnectionAttempts++ - 1])
+  );
 
-  vncClient = new VncClient(vmConsoleContainer.value, url.toString(), {
+  vncClient = new VncClient(vmConsoleContainer.value!, url.value!.toString(), {
     wsProtocols: ["binary"],
   });
-
   vncClient.scaleViewport = true;
+
+  vncClient.addEventListener("disconnect", handleDisconnectionEvent);
+  vncClient.addEventListener("connect", handleConnectionEvent);
+};
+
+watch(url, clearVncClient);
```

Comment:
Why this change?

Reply:
When switching from one VM console to another, the component is not unmounted. (probably a behavior of `vue`.)
This `watch` is therefore used to clear the connection to the vnc client each time the VM change.